Source/WebCore/platform/graphics/filters/FilterOperation.cpp

```diff
@@ -181,10 +177,8 @@ bool BasicColorMatrixFilterOperation::transformColor(SRGBA<float>& color) const
 
 inline bool BasicColorMatrixFilterOperation::operator==(const FilterOperation& operation) const
 {
-    if (!isSameType(operation))
-        return false;
-    const BasicColorMatrixFilterOperation& other = downcast<BasicColorMatrixFilterOperation>(operation);
-    return m_amount == other.m_amount;
+    auto* basicOperation = dynamicDowncast<BasicColorMatrixFilterOperation>(operation);
```

Comment:
I don't think this is equivalent to the previous code at all.
isSameType() was checking that the actually m_type variable was equal. A given class may use several types. This is the case for BasicColorMatrixFilterOperation for example. Your new check does:
```
    static bool isBasicColorMatrixFilterOperationType(Type type)
    {
        return type == Type::Grayscale || type == Type::Sepia || type == Type::Saturate || type == Type::HueRotate;
    }
```

So the type can now be any of these. Instead of being exactly the same type as the other `==` operand.